miso-web/src/spring-test/java/uk/ac/bbsrc/tgac/miso/webapp/springtest/LibraryDesignRestControllerST.java

```diff
@@ -0,0 +1,113 @@
+package uk.ac.bbsrc.tgac.miso.webapp.springtest;
+
+import org.junit.Test;
+
+import org.springframework.security.test.context.support.WithMockUser;
+import static org.junit.Assert.*;
+import java.util.List;
+import java.util.ArrayList;
+
+import uk.ac.bbsrc.tgac.miso.dto.Dtos;
+import uk.ac.bbsrc.tgac.miso.core.data.LibraryDesign;
+import uk.ac.bbsrc.tgac.miso.dto.LibraryDesignDto;
+
+
+
+public class LibraryDesignRestControllerST extends AbstractST {
+
+  private static final String CONTROLLER_BASE = "/rest/librarydesigns";
+  private static final Class<LibraryDesign> entityClass = LibraryDesign.class;
+
+  private List<LibraryDesignDto> makeCreateDtos() {
+
+    List<LibraryDesignDto> dtos = new ArrayList<LibraryDesignDto>();
+    LibraryDesignDto one = new LibraryDesignDto();
+    one.setName("1");
+    one.setSampleClassId(1L);
+    one.setStrategyId(1L);
+    one.setSelectionId(1L);
+    one.setDesignCodeId(1L);
+
+    LibraryDesignDto two = new LibraryDesignDto();
+    two.setName("2");
+    two.setSampleClassId(23L);
+    two.setStrategyId(2L);
+    two.setSelectionId(2L);
+    two.setDesignCodeId(2L);
+
+    dtos.add(one);
+    dtos.add(two);
+
+    return dtos;
+  }
+
+  @Test
+  @WithMockUser(username = "admin", password = "admin", roles = {"INTERNAL", "ADMIN"})
+  public void testBulkCreateAsync() throws Exception {
+    List<LibraryDesign> codes = baseTestBulkCreateAsync(CONTROLLER_BASE, entityClass, makeCreateDtos());
+    assertEquals("1", codes.get(0).getName());
+    assertEquals(1L, codes.get(0).getSampleClass().getId());
+    assertEquals(1L, codes.get(0).getLibraryStrategyType().getId());
+    assertEquals(1L, codes.get(0).getLibrarySelectionType().getId());
+    assertEquals(1L, codes.get(0).getLibraryDesignCode().getId());
+
+    assertEquals("2", codes.get(1).getName());
+    assertEquals(23L, codes.get(1).getSampleClass().getId());
+    assertEquals(2L, codes.get(1).getLibraryStrategyType().getId());
+    assertEquals(2L, codes.get(1).getLibrarySelectionType().getId());
+    assertEquals(2L, codes.get(1).getLibraryDesignCode().getId());
```

Comment:
Using linked objects that all have the same ID isn't ideal as you wouldn't be detecting if anything got crossed. There could be a bug in the code switching library selection and library strategy, for example.